### DISCUSSION TITLE
transmiteNotes aka kMIDIPropertyTransmitsNotes

### DISCUSSION THREAD

**Discussion by arshiacont** [orchetect/swift-midi#205] [Q&A] [answered]
I was wondering whether there is a way to safely realize if a MIDI Endpoint is capable of transmitting Note On/Off messages? 
I can see that the value of `transmiteNotes` is false for some MIDI Keyboards! Is this because the hardware does not provide correctly the spec? A workaround?

Goal here is to distinguish between a MIDI Instrument and a controller (or Virtual Endpoint that is either).

**Comment by eNorthug** [comment]
Hi, Arshia
I'm no expert, but believe what Endpoint callbacks are not part of the MIDI
spec or normal sequencer features. However, the ability to toggle MIDI
transmission (MIDI thru on/off) is part of every MIDI controllers user
options. Hope this was useful.
Best regards
Even N.

**Comment by orchetect** [comment, marked as answer]
The short answer is "not really."

Core MIDI, when it was designed 30 some-odd years ago included endpoint properties with the hopes that they would all be set by manufacturers. But in reality, few if any do. You're able to modify some of those properties in Audio MIDI Setup as an end-user as well, as part of setting up your "MIDI studio." But like manufacturers, users rarely if ever bother to do that either. We're all lazy and want to just plug-and-play 🤷🏻‍♂️.

So even though many properties are available on endpoints, most are fairly useless or outright unreliable.

That said, there have been some MIDI handshake messages that were developed over the years to try and come up with a standard for asking a device what it is capable of. You can Google 'MIDI Device Inquiry' messages (MIDI 1.0). This is something that a piece of hardware  or a virtual MIDI port would send and/or respond to, as well as software such as Cubase will send these messages to try and gain information about an attached device. But again, not all hardware and software use or understand these messages. So you have to operate under the assumption that most won't, but if you get a response then it's a bonus.

MIDI 2.0 has a new MIDI Capability Inquiry (MIDI-CI) spec that improves on the Device Inquiry concept and makes it more official, but MIDI 2.0 adoption is sparse at best at the current time.
